src/plugins/sync-plugin.js

```diff
@@ -194,14 +194,18 @@ const restoreRelativeSelection = (tr, relSel, binding) => {
     const anchor = relativePositionToAbsolutePosition(binding.doc, binding.type, relSel.anchor, binding.mapping)
     const head = relativePositionToAbsolutePosition(binding.doc, binding.type, relSel.head, binding.mapping)
     if (anchor !== null && head !== null) {
-      tr = tr.setSelection(TextSelection.create(tr.doc, anchor, head))
+      const selection = relSel.isNodeSelection
+        ? NodeSelection.create(tr.doc, Math.min(anchor, head))
+        : TextSelection.create(tr.doc, anchor, head)
+      tr = tr.setSelection(selection)
     }
   }
 }
 
 export const getRelativeSelection = (pmbinding, state) => ({
   anchor: absolutePositionToRelativePosition(state.selection.anchor, pmbinding.type, pmbinding.mapping),
-  head: absolutePositionToRelativePosition(state.selection.head, pmbinding.type, pmbinding.mapping)
+  head: absolutePositionToRelativePosition(state.selection.head, pmbinding.type, pmbinding.mapping),
+  isNodeSelection: state.selection instanceof NodeSelection,
```

Comment:
Can you please change this to `type: 'NodeSelection' | 'TextSelection'`? This would make it possible to potentially add more types of selections.